src/main/java/com/zendesk/maxwell/producer/MaxwellKinesisProducer.java

```diff
@@ -0,0 +1,131 @@
+
+import com.zendesk.maxwell.MaxwellContext;
+import com.zendesk.maxwell.replication.BinlogPosition;
+import com.zendesk.maxwell.row.RowMap;
+
+import com.amazonaws.services.kinesis.producer.Attempt;
+import com.amazonaws.services.kinesis.producer.KinesisProducer;
+import com.amazonaws.services.kinesis.producer.KinesisProducerConfiguration;
+import com.amazonaws.services.kinesis.producer.UserRecordFailedException;
+import com.amazonaws.services.kinesis.producer.UserRecordResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class KinesisCallback implements FutureCallback<UserRecordResult> {
+	public static final Logger logger = LoggerFactory.getLogger(KinesisCallback.class);
+
+	private InflightMessageList inflightMessages;
+	private final MaxwellContext context;
+	private final BinlogPosition position;
+	private final boolean isTXCommit;
+	private final String json;
+	private final String key;
+
+	public KinesisCallback(InflightMessageList inflightMessages, BinlogPosition position, boolean isTXCommit, MaxwellContext context, String key, String json) {
+		this.inflightMessages = inflightMessages;
+		this.context = context;
+		this.position = position;
+		this.isTXCommit = isTXCommit;
+		this.key = key;
+		this.json = json;
+	}
+
+	@Override
+	public void onFailure(Throwable t) {
+		logger.error(t.getClass().getSimpleName() + " @ " + position + " -- " + key);
+		logger.error(t.getLocalizedMessage());
+
+		if(t instanceof UserRecordFailedException) {
+			Attempt last = Iterables.getLast(((UserRecordFailedException) t).getResult().getAttempts());
+			logger.error(String.format("Record failed to put - %s : %s", last.getErrorCode(), last.getErrorMessage()));
+		}
+
+		logger.error("Exception during put", t);
+
+		markCompleted();
+	};
+
+	@Override
+	public void onSuccess(UserRecordResult result) {
+		if(logger.isDebugEnabled()) {
+			logger.debug("->  key:" + key + ", shard id:" + result.getShardId() + ", sequence number:" + result.getSequenceNumber());
+			logger.debug("   " + json);
+			logger.debug("   " + position);
+			logger.debug("");
+		}
+
+		markCompleted();
+	};
+
+	private void markCompleted() {
+		if(isTXCommit) {
+			BinlogPosition newPosition = inflightMessages.completeMessage(position);
+
+			if(newPosition != null) {
+				context.setPosition(newPosition);
+			}
+		}
+	}
+}
+
+public class MaxwellKinesisProducer extends AbstractProducer {
+	private static final Logger logger = LoggerFactory.getLogger(MaxwellKinesisProducer.class);
+
+	private final InflightMessageList inflightMessages;
+
+	private final KinesisProducer kinesisProducer;
+	private final String kinesisStream;
+
+	public MaxwellKinesisProducer(MaxwellContext context, String kinesisStream) {
+		super(context);
+
+		this.inflightMessages = new InflightMessageList();
+
+		this.kinesisStream = kinesisStream;
+
+		KinesisProducerConfiguration config = KinesisProducerConfiguration.fromPropertiesFile("kinesis-producer-library.properties");
```

Comment:
will this crash if the file doesn't exist?  Is it valid to simply run kinesis without extra configuration?